hawkbit-simple-ui/src/main/java/org/eclipse/hawkbit/ui/simple/view/util/TableView.java

```diff
@@ -45,15 +45,16 @@ public TableView(
             final BiFunction<Query<T, Void>, String, Stream<T>> queryFn,
             final Function<SelectionGrid<T, ID>, CompletionStage<Void>> addHandler,
             final Function<SelectionGrid<T, ID>, CompletionStage<Void>> removeHandler) {
-        this(rsql, null, entityRepresentation, queryFn, addHandler, removeHandler);
+        this(rsql, null, entityRepresentation, queryFn, addHandler, removeHandler, null);
     }
 
     public TableView(
             final Filter.Rsql rsql, final Filter.Rsql alternativeRsql,
             final SelectionGrid.EntityRepresentation<T, ID> entityRepresentation,
             final BiFunction<Query<T, Void>, String, Stream<T>> queryFn,
             final Function<SelectionGrid<T, ID>, CompletionStage<Void>> addHandler,
-            final Function<SelectionGrid<T, ID>, CompletionStage<Void>> removeHandler) {
+            final Function<SelectionGrid<T, ID>, CompletionStage<Void>> removeHandler,
+            final Function<SelectionGrid<T, ID>, CompletionStage<Void>> assignHandler) {
```

Comment:
Should assign handler be handled at TableView level? Table view is just a generic table with optional add/remove. Why not add the reset of the actions (buttons) in the concrete views (e.g. TargetView).
Even if not possible, maybe at least the TableView (addRemoveControls) shall be extended with something like adding set of additional actions (one of which will be assignHandler).

Reply:
assign action is now TargetView specific.

Reply:
thanks. Seems better now. @denislavprinov will review it again